### ISSUE TITLE
INCOMING: ERR_INVALID_PACKET_SIZE

### ISSUE THREAD

**Issue by rstefko** [kittoku/Open-SSTP-Client#134]
Trying to get the app running with MikroTik. I use my own certificate using IP address. Verify Hostname is disabled and custom Trusted Certificates folder is used.

PAP auth is enabled on MikroTik. Auth works but after that I immediatelly receive the following error:

```
[2024-06-10 15:34:38.285] Establish VPN connection
[2024-06-10 15:34:39.888] INCOMING: ERR_INVALID_PACKET_SIZE
[2024-06-10 15:34:39.901] Terminate VPN connection
```

I have attached debugger to the Android Studio. The variable **size** in this case is 28519 bytes. I will continue debugging, but any advice would be appretiated.

**Comment by rstefko** [comment]
It happens only with PAP auth. With MS-CHAPv2 connection works.

**Comment by rstefko** [comment]
It gives me `08 4C 6F 67 69 6E 20 6F 6B`, so something like "Login ok"?

**Comment by kittoku** [comment]
Maybe it is a fragmented [Authenticate-Ack](https://datatracker.ietf.org/doc/html/rfc1334#section-2.2.2). `08` seems to be  Msg-Length.

**Comment by kittoku** [comment]
Thank you for your investigation, @rstefko. So MikroTik produces Authenticate-Acks which DO NOT follow RFC(`Length` == 4, but `Msg-Length` and `Message` are still included)?

**Comment by khashayarnazari** [comment]
ok

دریافت Outlook برای Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by khashayarnazari** [comment]
ok

دریافت Outlook برای Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by rstefko** [comment]
Yes, message with code 2 is sent if that is what you ask. I am not sure why `Msg-Length` and `Message` is problem, they are also mentioned in the [RFC](https://datatracker.ietf.org/doc/html/rfc1334#section-2.2.2).

**Comment by khashayarnazari** [comment]
2
ok

دریافت Outlook برای Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by kittoku** [comment]
Of cource, `Msg-Length` and `Message` are accepted, but `Length` must be equal to the whole packet size. In your case, `Length` should be 13(=4+1+8). This limitation, which is defined in RFC, is necessary to parse packetes properly.

Maybe I could modify the souce codes for MicroTik, but it will make a little bit complicated codes. Also, the side effects should be carefully concerened. Futhermore, you already showed a workaroud of using MS-CHAPv2 (which is more secure than PAP!). So I'm not much motivated to make some changes. 

I recommend that you report this bug to MicroTik developers.
